lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1844,19 +1845,30 @@ instance NFData ChimericAccount
 data DelegationCertificate
     = CertDelegateNone ChimericAccount
     | CertDelegateFull ChimericAccount PoolId
+    | CertRegisterKey ChimericAccount
     deriving (Generic, Show, Eq, Ord)
 
 instance NFData DelegationCertificate
 
+data StakeKeyCertificate
+    = StakeKeyRegistration
+    | StakeKeyDeregistration
+    deriving (Generic, Show, Read, Eq)
```

Comment:
There are so many different kinds of certificate types in here already, for multiple purposes. I'd preferred to keep it closer to the DB.